pkg/controller/notification/mailgun_notification_delivery_service_test.go

```diff
@@ -127,4 +128,21 @@ func TestMailgunNotificationDeliveryService(t *testing.T) {
 		require.Error(t, err)
 		require.Equal(t, "template: template:1: function \"invalid_expression\" not defined", err.Error())
 	})
+
+	t.Run("test mailgun notification delivery service send with reply-to address", func(t *testing.T) {
+		// when
+		config := NewNotificationDeliveryServiceFactoryConfig("mg.foo.com", "abcd12345",
+			"noreply@foo.com", "info@foo.com", "mailgun")
+
+		mgds := NewMailgunNotificationDeliveryService(config, templateLoader, mockServerOption)
+		err := mgds.Send(notCtx, &v1alpha1.Notification{
+			Spec: v1alpha1.NotificationSpec{
+				Subject: "test",
+				Content: "abc",
+			},
+		})
+
+		// then
+		require.NoError(t, err)
```

Comment:
I'm afraid that the test doesn't verify anything. It's apparent because - as I stated in the previous comment - the reply-to email wasn't used anywhere. 
Could you please verify it either via the mock or using [Gock](https://github.com/h2non/gock)

Reply:
Unfortunately the mock doesn't provide us the capability to examine events created, it instead provides an events API that returns a random array of various Mailgun events - see https://github.com/mailgun/mailgun-go/blob/master/mock_events.go#L12.  This means the best we can do without actually testing with a real Mailgun API key is to just ensure that the Mailgun service doesn't return an error when using a mock server.

Reply:
> You could try to use the Gock mocking library to check the outcoming requests, but if you feel confident with verifying it via the mocked Mailgun service, then I'm fine with it.

All the existing Mailgun features were verified this way and tested in staging.  @alexeykazakov will we have the opportunity to test this in staging also before it gets deployed to production?

Reply:
No, it won't - everything what is deployed to staging is at the same time also deployed to prod.
I'm not afraid of breaking it now (in the sense that I don't think that we would break it), I was more thinking about adding a test to verify that it won't be unintentionally changed in a wrong way in the future. However, if the effort/time to implement such a test would be much bigger than the benefit from checking that it's sending the correct request, then it makes sense to omit it.

Reply:
I'm going to do some local testing with this today before I merge, using the live Mailgun API keys.  This will at least confirm that the ReplyTo is working as intended before we deploy to prod.